Code/Framework/AzCore/AzCore/IO/Path/Path.inl

```diff
@@ -185,20 +198,26 @@ namespace AZ::IO::Internal
     template <typename InputIt, typename EndIt, typename = AZStd::enable_if_t<AZStd::Internal::is_input_iterator_v<InputIt>>>
     static constexpr bool IsAbsolute(InputIt first, EndIt last, const char preferredSeparator)
     {
-        size_t pathSize = AZStd::distance(first, last);
-
         // If the preferred separator is a forward slash
-        // than an absolute path is simply one that starts with a forward slash
-        if (preferredSeparator == '/')
-        {
+        // than an absolute path is simply one that starts with a forward slash,
+        // unless we're on a posix platform that uses a custom path root separator
+        if (preferredSeparator == PosixPathSeparator)
+        {
+        #if defined(AZ_TRAIT_CUSTOM_PATH_ROOT_SEPARATOR)
+            const AZStd::string_view path{ first, last };
+            return path.find(AZ_TRAIT_CUSTOM_PATH_ROOT_SEPARATOR) != AZStd::string_view::npos;
+        #else
+            const size_t pathSize = AZStd::distance(first, last);
             return pathSize > 0 && IsSeparator(*first);
+        #endif
         }
         else
         {
             if (Internal::HasDrivePrefix(first, last))
             {
                 // If a windows path ends starts with C:foo it is a root relative path
                 // A path is absolute root absolute on windows if it starts with <drive_letter><colon><path_separator>
+                const size_t pathSize = AZStd::distance(first, last);
```

Comment:
I don't think pathSize is necessary, if its less than 2, it will fail ::HasDrivePrefix anyways

Reply:
Just moving it from above and keeping the existing logic the same, but if you want me to change it while I'm here I can. 
@lumberyard-employee-dm ?

Reply:
@spham-amzn Actually, the `pathSize` is necessary here.
`HasDrivePrefix` with iterators representing a path of "C:" will return true.

Reply:
To clarify for Windows paths `C:` **is NOT an absolute path**, but `C:\` is absolute path

Reply:
Actually you are right. The path length is to short circuit out the return so that the Internal::IsSeparator(..) is not called on anything less that 3 character